### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by BoikoMykola**
homework2 Boiko Mykola.

### CODE REVIEW

**Comment by BoikoMykola** [comment]
Check please.

**Comment by BoikoMykola** [comment]
Everything works now. Check, please.

File: homework2/sass/common.scss [hunk 1]
```diff
@@ -0,0 +1,54 @@
+
```
Comment (Maxim-Bondar) [inline homework2/sass/common.scss]: remove empty lines

File: homework2/sass/common.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+$color-mark: #E5E5E5;
+
+
+@mixin fontProp($fs,$lh,$c) {
+  font-size: $fs;
+  line-height: $lh;
+  color: $c;
+}
+
+.isHidden{
```
Comment (Maxim-Bondar) [inline homework2/sass/common.scss]: class names should be in kebab-case

File: homework2/sass/common.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
+
+
+$color-secondary-text: #7F7F7F;
```
Comment (Maxim-Bondar) [inline homework2/sass/common.scss]: variables and mixins should split into different files

File: homework2/sass/common.scss [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  font-size: $fs;
+  line-height: $lh;
+  color: $c;
+}
+
+.isHidden{
+  display: none;
+}
+
+@mixin tpl-button($border-color: $color-link, $font-color: $color-link, $bgc: $color-primary, $btn-width: 55px,
```
Comment (Maxim-Bondar) [inline homework2/sass/common.scss]: what does `tp1-button` means? also bad idea to set link related variables as default button property value
Also such mixin is to complex, and I guess you can just create just a simple class for button

File: homework2/sass/footer.scss [hunk 5]
```diff
@@ -0,0 +1,200 @@
+@import "common";
+
+footer {
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: use class instead of tag selector

File: homework2/sass/footer.scss [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+              display: flex;
+              flex-direction: column;
+              text-transform: uppercase;
+              @include fontProp(12px, 24px, $color-secondary-text);
+              letter-spacing: 2px;
+              padding-left: 6px;
+
+              a {
+                padding: 0 0 6px;
+                @include fontProp(16px, 24px, $color-secondary-text);
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: you dublicate properties from mixin, but you only need just change a font-size, so don't need to use such mixin if you only need to change one property

File: homework2/sass/footer.scss [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+                }
+              }
+            }
+          }
+        }
+      }
+
+      .contact-container {
+        .contact {
+          border-top: 2px solid $color-link;
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: Why do you use link relatet variable for `border-top`?

File: homework2/sass/footer.scss [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+          padding: 8px 23px;
+
+          > :first-child {
+            @include fontProp(16px, 36px, $color-secondary-text);
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss:56]: the same as was mntioned above for such mixin

File: homework2/sass/footer.scss [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+          }
+        }
+      }
+
+      .contact-container {
+        .contact {
+          border-top: 2px solid $color-link;
+          padding: 8px 23px;
+
+          > :first-child {
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: not valid selector , if you want to change first `.contact` element, selector should be `&:first-child`
If another purpose , just create a class

File: homework2/sass/footer.scss [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+            @include fontProp(16px, 36px, $color-secondary-text);
+          }
+
+          > :nth-child(2) {
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: the same as above

File: homework2/sass/footer.scss [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+            display: block;
+            padding-bottom: 4px;
+            @include fontProp(16px, 24px, $color-secondary-text);
+
+            &:hover {
+              @include fontProp(18px, 24px, #333);
+            }
+          }
+
+          div {
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: use class instead

File: homework2/sass/footer.scss [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+            @include fontProp(12px, 20px, $color-secondary-text);
+          }
+
+          > :last-child {
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: such kind of selector is not valid, read comment above

File: homework2/sass/footer.scss [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+      }
+    }
+  }
+}
+
+@media (min-width: 1440px) {
+  footer {
+    .wrapper {
+      padding: 78px 135px 69px 135px;
+      max-width: 1440px;
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: such general styles was created in .container class, don't need to dublicate, Also read about DRY principle

File: homework2/sass/footer.scss [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+      margin: 0 auto;
+
+      .container {
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: it is a general class , so set a different class name

File: homework2/sass/footer.scss [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+        display: flex;
+        justify-content: space-between;
+
+        nav {
```
Comment (Maxim-Bondar) [inline homework2/sass/footer.scss]: use class instead tag

File: homework2/sass/header.scss [hunk 16]
```diff
@@ -0,0 +1,343 @@
+@import "common";
+
+header {
```
Comment (Maxim-Bondar) [inline homework2/sass/header.scss]: class selector instead of tag

File: homework2/sass/header.scss [hunk 17]
```diff
@@ -0,0 +1,343 @@
...
+  font-family: 'Acme', sans-serif;
+  position: relative;
+  width: 100%;
+  min-width: 375px;
```
Comment (Maxim-Bondar) [inline homework2/sass/header.scss]: redundent style

File: homework2/sass/header.scss [hunk 18]
```diff
@@ -0,0 +1,343 @@
...
+  z-index: 10;
+  background: linear-gradient(45deg, #12C1E6, #01F973);
+
+  .header-container {
+    width: 100%;
+    height: 62px;
+
+    .wrapper {
+      max-width: 1440px;
+      margin: 0 auto;
```
Comment (Maxim-Bondar) [inline homework2/sass/header.scss]: dublicate styles from .container

File: homework2/sass/header.scss [hunk 19]
```diff
@@ -0,0 +1,343 @@
...
+          display: block;
+          position: relative;
+          width: 30px;
+          height: 20px;
+          cursor: pointer;
+
+          span, &::before, &::after {
+            left: 0;
+            position: absolute;
+            height: 15%;
```
Comment (Maxim-Bondar) [inline homework2/sass/header.scss]: magic number

File: homework2/sass/header.scss [hunk 20]
```diff
@@ -0,0 +1,343 @@
...
+                    background-color: $color-link;
+                    display: block;
+                    text-align: center;
+                    transition: all 0.3s ease 0s;
+                  }
+                }
+              }
+
+
+              &.get-skout-btn, &.login-btn {
```
Comment (Maxim-Bondar) [inline homework2/sass/header.scss]: you don't need to create a new class for each button, the buttons on the page has only two states, so in most cases you just need a general class for common styles and uniqu class for each state which is just change a color , background etc.

File: homework2/sass/header.scss [hunk 21]
```diff
@@ -0,0 +1,343 @@
...
+    }
+  }
+
+  .header-container-fixed {
+    position: fixed;
+    background-color: $color-primary;
+
+    .menu-icon {
+      span, &::before, &::after {
+        background-color: $color-secondary-text !important;
```
Comment (Maxim-Bondar) [inline homework2/sass/header.scss]: don't use !important, the only case when you can use it, whwn you need override a plugin styles

File: homework2/sass/header.scss [hunk 22]
```diff
@@ -0,0 +1,343 @@
...
+        margin-top: 30px;
+        align-self: center;
+        @include tpl-button;
+        border: none;
+      }
+    }
+  }
+}
+
+@media (min-width: 1440px) {
```
Comment (Maxim-Bondar) [inline homework2/sass/header.scss]: why you use only one media-query ? what about 1200, 768?

File: homework2/script.js [hunk 23]
```diff
@@ -0,0 +1,72 @@
+const menuIcon = document.querySelector(".menu-icon");
+menuIcon.addEventListener("click", function () {
+    const navigate = document.getElementsByTagName("nav");
```
Comment (Maxim-Bondar) [inline homework2/script.js]: bad idea to get element by tag name, use class name or even id if your element is unique

File: homework2/index.html [hunk 24]
```diff
@@ -0,0 +1,464 @@
...
+<body>
+<!-- HEADER -->
+<header>
+    <div class="header-container">
+        <div class="wrapper">
+            <div class="logo">
+                <a href="#"><img id="logo" src="image/SKOUT.png" alt="SKOUT"></a>
+            </div>
+            <div class="header-menu">
+                <div class="menu-icon">
```
Comment (Maxim-Bondar) [inline homework2/index.html]: should be a button instead of div

File: homework2/index.html [hunk 25]
```diff
@@ -0,0 +1,464 @@
...
+        <button class="slider-btn">Get SKOUT</button>
+    </div>
+</div>
+
+<!-- PLANS -->
+<div class="container">
+    <div class="plans-block">
+        <h2 class="plans-title">SKOUT Plans</h2>
+        <div class="plans">
+            <div class="plan-item">
```
Comment (Maxim-Bondar) [inline homework2/index.html]: it is a ul list

File: homework2/index.html [hunk 26]
```diff
@@ -0,0 +1,464 @@
...
+            </div>
+            <div class="partner-item">
+                <img src="image/vayner.png" alt="vayner">
+            </div>
+            <div class="partner-item">
+                <img src="image/Bsron.png" alt="Bsron">
+            </div>
+
+        </div>
+<!--        <div class="partner-list"></div>-->
```
Comment (Maxim-Bondar) [inline homework2/index.html]: remove comment

File: homework2/index.html [hunk 27]
```diff
@@ -0,0 +1,464 @@
...
+                    <div>
+                        <p>IDA Business and Technology
+                            Park, Mountrath Road, Unit 2 Portlaoise, Ireland</p>
+                    </div>
+                    <a href="#">View Map</a>
+                </div>
+            </div>
+        </div>
+        <div class="social">
+            <div>
```
Comment (Maxim-Bondar) [inline homework2/index.html]: ul list instead

File: homework2/index.html [hunk 28]
```diff
@@ -0,0 +1,464 @@
...
+                <a href="#"><img src="image/twitter.png" alt="twitter"></a>
+                <a href="#"><img src="image/facebook.png" alt="facebook"></a>
+                <a href="#"><img src="image/youTube.png" alt="youTube"></a>
+                <a href="#"><img src="image/inst.png" alt="instagram"></a>
+            </div>
+            <img src="image/cyber.png" alt="skout">
```
Comment (Maxim-Bondar) [inline homework2/index.html]: logo must be a link

File: homework2/index.html [hunk 29]
```diff
@@ -0,0 +1,464 @@
...
+                    <li>
+                        <a href="#">Careers</a>
+                    </li>
+                    <li>
+                        Call the SOC
+                        <a href="tel:+18558384500">1-855-838-4500</a>
+                    </li>
+                </ul>
+            </nav>
+            <div class="contact-container">
```
Comment (Maxim-Bondar) [inline homework2/index.html]: it is also can be transformed to list

**Review by Maxim-Bondar — CHANGES_REQUESTED** [review]
COMMON
- use classes instead of tag selector
- create general class for button and don't need to use `tpl-button` mixin
- if you need override only one property don't need to use a mixin, in this way you avoid of  code dublicating
- all interactive elements must have a cursor pointer
- sticky menu is appear only when start scroll up

File: homework2/sass/common.scss (hunk 1 above)
Comment (BoikoMykola) [reply homework2/sass/common.scss]: done

File: homework2/sass/common.scss (hunk 2 above)
Comment (BoikoMykola) [reply homework2/sass/common.scss]: Done

File: homework2/sass/footer.scss (hunk 5 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/sass/footer.scss (hunk 14 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/sass/header.scss (hunk 16 above)
Comment (BoikoMykola) [reply homework2/sass/header.scss]: Done

File: homework2/sass/header.scss (hunk 17 above)
Comment (BoikoMykola) [reply homework2/sass/header.scss]: done

File: homework2/sass/header.scss (hunk 18 above)
Comment (BoikoMykola) [reply homework2/sass/header.scss]: Done

File: homework2/sass/header.scss (hunk 19 above)
Comment (BoikoMykola) [reply homework2/sass/header.scss]: Done

File: homework2/index.html (hunk 26 above)
Comment (BoikoMykola) [reply homework2/index.html]: Done

File: homework2/sass/common.scss (hunk 3 above)
Comment (BoikoMykola) [reply homework2/sass/common.scss]: Done

File: homework2/sass/common.scss (hunk 4 above)
Comment (BoikoMykola) [reply homework2/sass/common.scss]: "tpl-button" mean "typical-button".
Changed mixin to class.

File: homework2/sass/footer.scss (hunk 7 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/index.html (hunk 29 above)
Comment (BoikoMykola) [reply homework2/index.html]: done

File: homework2/index.html (hunk 28 above)
Comment (BoikoMykola) [reply homework2/index.html]: Done

File: homework2/index.html (hunk 27 above)
Comment (BoikoMykola) [reply homework2/index.html]: Done

File: homework2/index.html (hunk 25 above)
Comment (BoikoMykola) [reply homework2/index.html]: Done

File: homework2/index.html (hunk 24 above)
Comment (BoikoMykola) [reply homework2/index.html]: Done

File: homework2/script.js (hunk 23 above)
Comment (BoikoMykola) [reply homework2/script.js]: Done

File: homework2/sass/header.scss (hunk 21 above)
Comment (BoikoMykola) [reply homework2/sass/header.scss]: Done

File: homework2/sass/header.scss (hunk 20 above)
Comment (BoikoMykola) [reply homework2/sass/header.scss]: Done

File: homework2/sass/footer.scss (hunk 13 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/sass/footer.scss (hunk 11 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/sass/footer.scss (hunk 12 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/sass/footer.scss (hunk 9 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/sass/footer.scss (hunk 10 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/sass/footer.scss (hunk 8 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss:56]: Done

File: homework2/sass/footer.scss (hunk 15 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/sass/footer.scss (hunk 6 above)
Comment (BoikoMykola) [reply homework2/sass/footer.scss]: Done

File: homework2/sass/header.scss (hunk 22 above)
Comment (BoikoMykola) [reply homework2/sass/header.scss]: Done 1440, 1200, 992, 768.

**Comment by BoikoMykola** [comment]
NAV folds up when scrolling down and appears when scrolling up.

**Comment by BoikoMykola** [comment]
Fixed all bugs. Check please.

**Pull request closed without merging**